src/kdbindings/node.h

```diff
@@ -148,20 +148,22 @@ class ConstantNode : public NodeInterface<T>
     T m_value;
 };
 
-template<typename PropertyType>
+template<typename PropertyType, bool IsConst>
 class PropertyNode : public NodeInterface<PropertyType>
 {
 public:
-    explicit PropertyNode(Property<PropertyType> &property)
+    using PropertyReference = std::conditional_t<IsConst, const Property<PropertyType> &, Property<PropertyType> &>;
```

Comment:
I don't think we need this conditional type, we should be able to just always take `const Property<PropertyType> &`.
We're never writing to it after all.